### DISCUSSION TITLE
Allow custom integrations to associate themselves with the `client:only` directive

### DISCUSSION THREAD

**Discussion by RohanHart** [withastro/roadmap#911] [Proposal]
# Summary

`clientOnlyValues` is a static `Set` containing only official integrations. Allowing a custom integration to register themselves with this under some name would provide full `client:only` support. 

# Background & Motivation

Client-only rendering with `@analogjs/astro-angular` or any other custom integration will not work when there are multiple renderers in the Astro configuration.

Aside from the the officially supported renderes, as defined in the `clientOnlyValues`, there is a fallback to guessing based on the component extension. This will not work well for Angular as their default naming scheme will result in `component` and it makes no sense for a custom integration to call itself that.

# Proposal

 - When a custom integration provides a `clientEntrypoint` that includes registration of a `client:only` renderer under the integrations name.
 - Optionally, extend `AstroRenderer` to allow a short alias for the renderer so that the directive can be specified more simply, eg., `client:only="angular"`

**Comment by matthewp** [comment]
Seems reasonable to me.
